datumaro/plugins/data_formats/datumaro/base.py

```diff
@@ -190,6 +199,10 @@ def _load_items(self, parsed):
                 media = MediaElement(path=media_desc.get("path"))
                 self._media_type = MediaElement
 
+            if has_hashkey_file(self._rootpath):
```

Comment:
You are trying to invoke disk I/O for every `DatasetItem` (O(N)). This will make it super slow.

Reply:
```python
    def _load_impl(self, path: str) -> None:
        """Actual implementation of loading Datumaro format."""
        self._infos = self._load_infos(self._parsed_anns)
        self._categories = self._load_categories(self._parsed_anns)
        items = self._load_items(self._parsed_anns)
        self._items = self._load_hash_key(items)

    def _load_hash_key(self, items):
        if not has_hashkey_file(self._rootpath):
            return items

        hashkey_dict = parse_hashkey_file(self._rootpath)
        for item in items:
            item.annotations.append(hashkey_dict[(item.id, item.subset)])
        return items
```